### PR TITLE
Added sample and exposed itemsource for MvxExpandableTableViewSource

### PR DESCRIPTION
**Pull request by b099l3**
I have slightly modified MvxExpandableTableViewSource so that when the itemSource is bound _isCollapsed gets updated.
Exposed ItemSource in MvxExpandableTableViewSource so that _isCollapsed gets updated. 

 To allow for a sample to be created #34

### CODE REVIEW

**Comment by Cheesebaron** [comment]
Why the revert?

File: MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs [hunk 1]
```diff
@@ -10,13 +10,31 @@
...
-    public abstract class MvxExpandableTableViewSource : MvxTableViewSource
-    {
+    public abstract class MvxExpandableTableViewSource<TItemSource, TItem> : MvxTableViewSource where TItemSource : List<TItem>
+	{
         /// <summary>
         /// Indicates which sections are expanded.
         /// </summary>
         private bool[] _isCollapsed;
 
+		private IEnumerable<TItemSource> _itemsSource;
```
Comment (Cheesebaron) [inline MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: Could you please use spaces instead of tabs?

Comment (b099l3) [reply MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: Cool that should be it changed

**Comment by b099l3** [comment]
I reverted as I wasn't too sure how to get the latest changes onto my fork once it was behind. Realised I didn't need to do this now, sorry.

File: MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs [hunk 2]
```diff
@@ -10,73 +10,91 @@
     using System.Linq;
     using UIKit;
 
-    public abstract class MvxExpandableTableViewSource : MvxTableViewSource
-    {
+    public abstract class MvxExpandableTableViewSource<TItemSource, TItem> : MvxTableViewSource where TItemSource : List<TItem>
```
Comment (martijn00) [inline MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: This makes the MvxExpandableTableViewSource generic. Can we have a non generic too?

Comment (b099l3) [reply MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: I have added a non generic version, if you want to take a look.

Comment (martijn00) [reply MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: Look good i think.

Comment (b099l3) [reply MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: I'll add a sample for this too, in abit.

Comment (b099l3) [reply MvvmCross.iOS.Support/Views/MvxExpandableTableViewSource.cs]: Sample added using the non generic MvxExpandableTableViewSource.

**Comment by b099l3** [comment]
This should close #34

**Pull request merged**